src/odata-to-abstract-sql.ts

```diff
@@ -1469,17 +1469,19 @@ const generateShortAliases = (clientModel: AbstractSqlModel) => {
 
 	const getRelationships = (
 		relationships: AbstractSqlModel['relationships'] | Relationship,
+		nestedRelationships: string[] = [],
 	): string[] => {
 		const relationshipKeys = Object.keys(relationships);
-		const nestedRelationships = [];
 		for (const key of relationshipKeys) {
 			if (key !== '$') {
-				nestedRelationships.push(
-					getRelationships(relationships[key] as Relationship),
+				nestedRelationships.push(key);
+				getRelationships(
+					relationships[key] as Relationship,
+					nestedRelationships,
 				);
 			}
 		}
-		return relationshipKeys.concat(...nestedRelationships);
+		return nestedRelationships;
 	};
```

Comment:
I was thinking whether having something like:
```
const getRelationships = (
	relationships: AbstractSqlModel['relationships'] | Relationship,
): string[] => {
	const nestedRelationships:string[] = [];

	const getRelationshipsInner = (
		relationships: AbstractSqlModel['relationships'] | Relationship,
		nestedRelationships: string[],
	): string[] => {
		const relationshipKeys = Object.keys(relationships);
		for (const key of relationshipKeys) {
			if (key !== '$') {
				nestedRelationships.push(key);
				getRelationshipsInner(
					relationships[key] as Relationship,
					nestedRelationships,
				);
			}
		}
		return nestedRelationships;
	};

	return getRelationshipsInner(relationships, nestedRelationships);
};
```
would make it feel better, but I guess since this is an internal method, just adding a comment on how this is used should be enough.

Reply:
Or maybe even just renaming the method to `populateRelationships` or something :thinking: to reflect what it's now doing.
Sounds like a fast and readable to approach, what do you think?

Reply:
I've added a comment, I'd rather not do the recursive function in-line as it could affect performance and in fact I've also moved the existing function to module level as it might help a little further